pandas/core/groupby.py

```diff
@@ -3584,21 +3584,15 @@ def decons_obs_group_ids(comp_ids, obs_ids, shape, labels):
 
 
 def _indexer_from_factorized(labels, shape, compress=True):
-    if _int64_overflow_possible(shape):
-        indexer = np.lexsort(np.array(labels[::-1]))
-        return indexer
-
-    group_index = get_group_index(labels, shape, sort=True, xnull=True)
+    ids = get_group_index(labels, shape, sort=True, xnull=False)
 
-    if compress:
-        comp_ids, obs_ids = _compress_group_index(group_index)
-        max_group = len(obs_ids)
+    if not compress:
+        ngroups = (ids.size and ids.max()) + 1
```

Comment:
Both `idx.size` and `ids.max()` are integers, yes?

I find using `and` for integers rather confusing. Perhaps this could be: `(idx.max() if ids.size else 0) + 1`